src/core/ext/filters/rbac/rbac_service_config_parser.cc

```diff
@@ -750,6 +806,15 @@ void RbacConfig::RbacPolicy::Rules::JsonPostLoad(const Json&, const JsonArgs&,
     ValidationErrors::ScopedField field(errors, ".action");
     errors->AddError("unknown action");
   }
+  auto rbac_audit_condition =
+      static_cast<Rbac::AuditCondition>(audit_condition);
```

Comment:
I might not be understanding the code structure here, but do we need to check that it's valid before casting?
If we cast to `Rbac::AuditCondition` before checking, what would the value be for something that isn't in our enum?

Reply:
We can't really know whether it's valid before casting unless we assume 0 is `kNone` and etc. But this is exactly what `static_cast` will do for us.

IIUC, it's ok to do it in the current order as long as populate the `ValidationErrors` appropriately. The `rbac_action` is processed the same way above.

Reply:
I took your suggestion here but it looks a bit different since I changed the type of audit_condition to `Rbac::AuditCondition` and manually parsed it in post loading. PTAL.